### PR TITLE
Phase2 bughunt v6

### PR DESCRIPTION
**Pull request by Ronak-Searce**
- [ ] Empty block of space is shown upon editing the primary key after deleting a column from the table.(53)
- [ ] Save session form doesn't show Notes as compulsary field but on saving shows error.(66)
- [x] Add column not working in the columns tab(89).
- [x] Getting convert to foreign key option while editing foreign key.(93)

### CODE REVIEW

File: ui/src/app/components/object-detail/object-detail.component.html [hunk 1]
```diff
@@ -689,7 +689,7 @@ <h3 class="title">
                   >
                     <span>Drop Foreign Key</span>
                   </button>
-                  <button
+                  <!-- <button
```
Comment (shreyakhajanchi) [inline ui/src/app/components/object-detail/object-detail.component.html:692]: Why has this been commented if its intended to be removed?

File: ui/src/app/components/object-detail/object-detail.component.ts [hunk 2]
```diff
@@ -507,9 +511,13 @@ export class ObjectDetailComponent implements OnInit {
...
     })
     for (let i = 0; i < this.localTableData.length; i++) {
-      if (!currentPkColumns.includes(this.localTableData[i].spColName))
+      if (
+        !currentPkColumns.includes(this.localTableData[i].spColName) &&
+        this.localTableData[i].spColName !== ''
+      )
         this.pkColumnNames.push(this.localTableData[i].spColName)
     }
+    console.log(this.pkColumnNames)
```
Comment (shreyakhajanchi) [inline ui/src/app/components/object-detail/object-detail.component.ts:520]: remove the log

File: ui/src/app/components/object-detail/object-detail.component.html (hunk 1 above)
Comment (shreyakhajanchi) [inline ui/src/app/components/object-detail/object-detail.component.html:692]: Dropping a fk is not waiting for the user to click on save and dropping it instantly

File: ui/src/app/components/object-detail/object-detail.component.ts [hunk 3]
```diff
@@ -319,16 +323,16 @@ export class ObjectDetailComponent implements OnInit {
...
       .map((item) => item.srcColName)
       .indexOf(this.addedColumnName)
-    this.tableData[index].spColName = this.droppedColumns[addedRowIndex].spColName
-    this.tableData[index].spDataType = this.droppedColumns[addedRowIndex].spDataType
-    this.tableData[index].spOrder = -1
-    this.tableData[index].spIsPk = this.droppedColumns[addedRowIndex].spIsPk
-    this.tableData[index].spIsNotNull = this.droppedColumns[addedRowIndex].spIsNotNull
+    this.localTableData[index].spColName = this.droppedColumns[addedRowIndex].spColName
+    this.localTableData[index].spDataType = this.droppedColumns[addedRowIndex].spDataType
+    this.localTableData[index].spOrder = 3
```
Comment (shreyakhajanchi) [inline ui/src/app/components/object-detail/object-detail.component.ts:333]: why is the order being assigned as 3?

File: ui/src/app/components/object-detail/object-detail.component.ts (hunk 2 above)
Comment (Ronak-Searce) [reply ui/src/app/components/object-detail/object-detail.component.ts:520]: Updated

File: ui/src/app/components/object-detail/object-detail.component.html (hunk 1 above)
Comment (Ronak-Searce) [reply ui/src/app/components/object-detail/object-detail.component.html:692]: Updated

File: ui/src/app/components/object-detail/object-detail.component.ts (hunk 3 above)
Comment (Ronak-Searce) [reply ui/src/app/components/object-detail/object-detail.component.ts:333]: Updated

**Review by shreyakhajanchi — APPROVED** [review]
LGTM

**Pull request merged**